src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -248,20 +249,43 @@ export class Visualization {
     this.forceSim.updateRelationships(this.graph)
   }
 
-  private handleZoomClick(zoomType: ZoomType): void {
+  private handleZoomClick = (zoomType: ZoomType): void => {
     this.draw = true
     this.isZoomClick = true
 
     if (zoomType === ZoomType.IN) {
-      this.zoomBehavior.scaleBy(this.root, 1.3)
+      this.zoomTo(1.3)
     } else if (zoomType === ZoomType.OUT) {
-      this.zoomBehavior.scaleBy(this.root, 0.7)
+      this.zoomTo(0.7)
     } else if (zoomType === ZoomType.FIT) {
-      this.zoomToFitWholeGraph()
+      const scaleAndOffset = this.getZoomScaleFactorToFitWholeGraph()
+      if (scaleAndOffset) {
+        const { scale, centerPointOffset } = scaleAndOffset
+        // Do not zoom in more than zoom max scale for really small graphs
+        this.zoomTo(Math.min(scale, ZOOM_MAX_SCALE), {
```

Comment:
We could use `centerPointOffset` directly here right? No need to create the new object?

Reply:
Like so: `this.zoomTo(Math.min(scale, ZOOM_MAX_SCALE), centerPointOffset))`